src/SigninResponse.ts

```diff
@@ -29,31 +29,29 @@ export class SigninResponse {
 
     // set by ResponseValidator
     /** custom "state", which can be used by a caller to have "data" round tripped */
-    public state: unknown | undefined;
+    public state: unknown;
 
     // set by ResponseValidator
-    public profile: UserProfile;
-
-    public constructor(url?: string, delimiter = "#") {
-        const values = UrlUtils.parseUrlFragment(url, delimiter);
-
-        this.error = values.error;
-        this.error_description = values.error_description;
-        this.error_uri = values.error_uri;
-
-        this.code = values.code;
-        this.state_id = values.state;
-
-        this.id_token = values.id_token;
-        this.session_state = values.session_state;
-        this.access_token = values.access_token;
-        this.refresh_token = values.refresh_token;
-        this.token_type = values.token_type;
-        this.scope = values.scope;
-        this.expires_in = parseInt(values.expires_in);
-
-        this.state = undefined;
-        this.profile = {};
+    public profile: UserProfile = {};
+
+    public constructor(params: URLSearchParams) {
+        // URLSearchParams returns `null` for missing values, reconstruct it as a map
+        const values = new Map(params);
+        this.error = values.get("error");
+        this.error_description = values.get("error_description");
+        this.error_uri = values.get("error_uri");
+
+        this.code = values.get("code") ?? "";
+        this.access_token = values.get("access_token") ?? "";
```

Comment:
for maintainability it probably makes sense to add here a comment before the code block with the `?? ""`. Just in case others are wonder why no throw. Something like:

`ResponseValidator takes care about invalid values`